From e6a80f20b7917dafe0e11fa43a575a2d3bb0f953 Mon Sep 17 00:00:00 2001
From: Brian Behlendorf <behlendorf1@llnl.gov>
Date: Thu, 3 Dec 2015 10:37:06 -0800
Subject: [PATCH 03/18] Set tags for latent ec2 buildslaves

EC2LatentBuildSlave does not respect the tags argument to __init__()
when launching spot instances.  Apply the tags in _wait_for_instance()
identical to what is done for attaching volumes.

http://trac.buildbot.net/ticket/2903
---
 master/buildbot/buildslave/ec2.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/master/buildbot/buildslave/ec2.py b/master/buildbot/buildslave/ec2.py
index 191999ea6..e16b6372b 100644
--- a/master/buildbot/buildslave/ec2.py
+++ b/master/buildbot/buildslave/ec2.py
@@ -274,8 +274,6 @@ class EC2LatentBuildSlave(AbstractLatentBuildSlave):
         instance_id, image_id, start_time = self._wait_for_instance(
             reservation)
         if None not in [instance_id, image_id, start_time]:
-            if len(self.tags) > 0:
-                self.conn.create_tags(instance_id, self.tags)
             return [instance_id, image_id, start_time]
         else:
             log.msg('%s %s failed to start instance %s (%s)' %
@@ -398,6 +396,8 @@ class EC2LatentBuildSlave(AbstractLatentBuildSlave):
                 minutes // 60, minutes % 60, seconds)
             if len(self.volumes) > 0:
                 self._attach_volumes()
+            if len(self.tags) > 0:
+                self.conn.create_tags(self.instance.id, self.tags)
             return self.instance.id, image.id, start_time
         else:
             return None, None, None
-- 
2.14.3

